optimum/neuron/trainers.py

```
@@ -438,11 +454,12 @@ def training_step(self, model: torch.nn.Module, inputs: Dict[str, Union[torch.Te
                 use_bf16 = self.accelerator.state.mixed_precision == "bf16"
                 dtype = torch.bfloat16 if use_bf16 else torch.float32
                 loss = torch.tensor(0, dtype=dtype).to(xm.xla_device())
-            else:
-                loss = loss.detach()
+
+            # TODO: handle with num_items_in_batch
```

Maybe raise an exception here if num_items_in_batch is not None ?